### PR TITLE
OLS-488: Readiness probe - index loading check

### PR DESCRIPTION
**Pull request by onmete**
## Description

Add index check to readiness probe.

## Type of change

- [x] Refactor
- [x] New feature

### CODE REVIEW

File: ols/app/endpoints/health.py [hunk 1]
```diff
@@ -7,28 +7,27 @@
...
-#     "status": "unhealthy",
-#     "version": "1.0.0",
-#     "dependencies": {
-#         "database": "healthy",
-#         "externalApi": "unhealthy"
-#     }
-# }
+def index_is_ready() -> bool:
+    """Check if the index is loaded."""
+    if config._rag_index is None:
```
Comment (xrajesh) [inline ols/app/endpoints/health.py:18]: @onmete When there is no RAG content  configured , the server can proceed without RAG content . In that case the readiness can return false .

File: ols/app/models/models.py [hunk 2]
```diff
@@ -342,6 +342,35 @@ class HealthResponse(BaseModel):
     }
 
 
+class ReadinessResponse(BaseModel):
+    """Model representing a response to a readiness request.
+
+    Attributes:
+        ready: The readiness of the service.
+        reason: The reason for the readiness.
```
Comment (xrajesh) [inline ols/app/models/models.py:350]: @onmete do we want the reason field ? just thinking if we can keep in sync with liveness, unless there is a use for this.

Comment (onmete) [reply ols/app/models/models.py:350]: Telling why service is not ready sounds like a good reason to me. But I didn't find any required schema for these two endpoints, so I guess it is on our needs. Do you have concerns with this schema?

File: ols/app/endpoints/health.py (hunk 1 above)
Comment (onmete) [reply ols/app/endpoints/health.py:18]: Right, updated.

File: docs/openapi.json [hunk 3]
```diff
@@ -196,30 +196,12 @@
...
                             "application/json": {
                                 "schema": {
-                                    "$ref": "#/components/schemas/HealthResponse"
+                                    "$ref": "#/components/schemas/ReadinessResponse"
                                 }
                             }
                         }
                     }
                 }
-            },
```
Comment (tisnik) [inline docs/openapi.json:205]: are not HEAD HTTP methods used regularly, especially for liveness and readiness probes? @JoaoFula ?

**Review by tisnik — COMMENTED** [review]
in overall it looks ok

File: docs/openapi.json (hunk 3 above)
Comment (JoaoFula) [reply docs/openapi.json:205]: yeah, I believe so. Don't normally use it myself but that's their purpose.

**Pull request merged**